packages/react-reconciler/src/ReactFiberRoot.new.js

```diff
@@ -104,7 +105,7 @@ export function markRootSuspendedAtTime(
     root.lastSuspendedTime = expirationTime;
   }
 
-  if (expirationTime <= root.lastPingedTime) {
+  if (didCompleteRoot && expirationTime <= root.lastPingedTime) {
```

Comment:
I think what's wrong here is that we always work on updates before we work on pings, i.e. we always check for "unsuspended" level, at any priority, before we check for a pinged level.

So if we finish at a lower priority update level, that doesn't mean we already tried the higher priority ping level.

So `lastPingedTime` should only be cleared if we suspend the whole root at that exact level. If we suspend at a lower level, we still should go back and try the higher priority level. (Also taking into account that we could get pinged again at that same level in an interleaved event during the  render phase, in which case we should try yet again.)

Need to write some more test cases. I can take a crack at it once I wrap up the current thing I'm working on.